flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskIOMetricGroup.java

```diff
@@ -80,18 +76,25 @@ public Meter getNumBytesOutRateMeter() {
 		return numBytesOutRate;
 	}
 
-	public MetricGroup getBuffersGroup() {
-		return buffers;
-	}
-
 	// ------------------------------------------------------------------------
 	//  metrics of Buffers group
 	// ------------------------------------------------------------------------
 
 	/**
+	 * initialize Buffer Metrics for a task
+	 */
+	public void initializeBufferMetrics(Task task) {
+		final MetricGroup buffers = addGroup("buffers");
+		buffers.gauge("inputQueueLength", new InputBuffersGauge(task));
+		buffers.gauge("outputQueueLength", new OutputBuffersGauge(task));
+		buffers.gauge("inPoolUsage", new InputBufferPoolUsageGauge(task));
+		buffers.gauge("outPoolUsage", new OutputBufferPoolUsageGauge(task));
+	}
+
+	/**
 	 * Input received buffers gauge of a task
 	 */
-	public static final class InputBuffersGauge implements Gauge<Integer> {
+	private final class InputBuffersGauge implements Gauge<Integer> {
```

Comment:
InputBuffersGauge is unaccessible for any class except TaskIOMetricGroup, so I change the public to private. It can work no matter whether it is static or not. to save outclass reference, static is prefer.